slashing-protection/src/test/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidatorTest.java

```diff
@@ -64,6 +69,39 @@ void verifyCachedGVRIsUsedForNewGVR(final Jdbi jdbi) {
     verify(metadataDao, times(1)).insertGenesisValidatorsRoot(any(), eq(gvr));
   }
 
+  @Test
+  void verifyCachedGVRReturnsTrueFromMultipleThreads(final Jdbi jdbi) throws InterruptedException {
+    final GenesisValidatorRootValidator gvrValidator =
+        new GenesisValidatorRootValidator(jdbi, metadataDao);
+    final Bytes32 gvr = Bytes32.leftPad(Bytes.of(3));
+
+    final int numberOfThreads = 10;
+    final ExecutorService executorService = Executors.newFixedThreadPool(numberOfThreads);
+    final CountDownLatch latch = new CountDownLatch(numberOfThreads);
+    final AtomicBoolean allTrue = new AtomicBoolean(true); // To keep track of the result
+
+    for (int i = 0; i < numberOfThreads; i++) {
+      executorService.submit(
+          () -> {
+            try {
+              // If any thread returns false, set allTrue to false
+              if (!gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(gvr)) {
+                allTrue.set(false);
+              }
+            } finally {
+              latch.countDown(); // Ensure the latch is counted down even if an exception is thrown
+            }
+          });
+    }
+
+    latch.await(); // Wait for all threads to finish
+    executorService.shutdown(); // Shutdown the executor service
+
+    assertThat(allTrue.get()).isTrue();
```

Comment:
Don't think this check or the latch is doing anything. 

When this fails, it's due to the number of times findGenesisValidatorsRoot is called

Reply:
`latch.await();` to make sure that all threads have joined. Agreed with allTrue. I added findByGVR assertions later on :). Will be removing allTrue assertion.

Reply:
decided to rename allTrue and still kept it.